src/Common/PoolWithFailoverBase.h

```diff
@@ -73,19 +73,14 @@ class PoolWithFailoverBase : private boost::noncopyable
     {
         TryResult() = default;
 
-        void reset()
-        {
-            entry = Entry();
-            is_usable = false;
-            is_up_to_date = false;
-            delay = 0;
-        }
+        void reset() { *this = {}; }
 
         Entry entry; /// use isNull() to check if connection is established
         bool is_usable = false; /// if connection is established, then can be false only with table check
                                 /// if table is not present on remote peer, -> it'll be false
         bool is_up_to_date = false; /// If true, the entry is a connection to up-to-date replica
                                     /// Depends on max_replica_delay_for_distributed_queries setting
+        bool is_readonly = false;   /// Table is in read-only mode, INSERT can ignore such replicas.
```

Comment:
Let's add the new field to the end of the structure, so the field will be declared in the same sequence as their serialization.